### PR TITLE
New Rule: Windows DNS Client Rule command 

### PR DESCRIPTION
**Pull request by bornatalebi**

### CODE REVIEW

**Comment by frack113** [comment]
It's works too 
```
PS C:\WINDOWS\system32> Add-DnsClientNrptRule -Namesp "test.test" -NameSe "10.0.0.1"
PS C:\WINDOWS\system32> Get-DnsClientNrptRule


Name                             : {BD0629C2-E6C0-48D6-A78E-D23DE63BC36B}
Version                          : 2
Namespace                        : {test.test}
IPsecCARestriction               :
DirectAccessDnsServers           :
DirectAccessEnabled              : False
DirectAccessProxyType            :
DirectAccessProxyName            :
DirectAccessQueryIPsecEncryption :
DirectAccessQueryIPsecRequired   :
NameServers                      : 10.0.0.1
DnsSecEnabled                    : False
DnsSecQueryIPsecEncryption       :
DnsSecQueryIPsecRequired         :
DnsSecValidationRequired         :
NameEncoding                     : Disable
DisplayName                      :
Comment                          :



PS C:\WINDOWS\system32>
```

**Comment by bornatalebi** [comment]
> It's works too
> 
> ```
> PS C:\WINDOWS\system32> Add-DnsClientNrptRule -Namesp "test.test" -NameSe "10.0.0.1"
> PS C:\WINDOWS\system32> Get-DnsClientNrptRule
> 
> 
> Name                             : {BD0629C2-E6C0-48D6-A78E-D23DE63BC36B}
> Version                          : 2
> Namespace                        : {test.test}
> IPsecCARestriction               :
> DirectAccessDnsServers           :
> DirectAccessEnabled              : False
> DirectAccessProxyType            :
> DirectAccessProxyName            :
> DirectAccessQueryIPsecEncryption :
> DirectAccessQueryIPsecRequired   :
> NameServers                      : 10.0.0.1
> DnsSecEnabled                    : False
> DnsSecQueryIPsecEncryption       :
> DnsSecQueryIPsecRequired         :
> DnsSecValidationRequired         :
> NameEncoding                     : Disable
> DisplayName                      :
> Comment                          :
> 
> 
> 
> PS C:\WINDOWS\system32>
> ```

So it should be `-Namesp*` and `-NameSe*` to cover all scenarios. Do I have to use wildcard(*) or not? Because apparently `contains` does that anyway.
According to de the wiki:
`contains`: puts `*` wildcards around the values, such that the value is matched anywhere in the field.

**Comment by frack113** [comment]
```yaml
        ScriptBlockText|contains|all:
            - 'Add-DnsClientNrptRule'
            - '-Namesp'
            - '-NameSe'
 ```
As it is a contains|all it will look for `*Add-DnsClientNrptRule*` AND `*-Namesp*` AND `*-NameSe*`

File: rules/windows/powershell/powershell_script/posh_ps_add_dnsclient_rule.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
+title: Windows DNS Client Commands
```
Comment (frack113) [inline rules/windows/powershell/powershell_script/posh_ps_add_dnsclient_rule.yml:1]: I think the title is too generic agains the rule detection.
Somethink like "Powershell Add Name Resolution Policy Table rule" ?

**Review by frack113 — APPROVED** [review]

**Pull request merged**